x/feegrant/keeper/keeper.go

```diff
@@ -52,8 +53,12 @@ func (k Keeper) GrantAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress,
 	key := feegrant.FeeAllowanceKey(granter, grantee)
 
 	var oldExp *time.Time
-	existingGrant, err := k.getGrant(ctx, grantee, granter)
-	if err != nil && existingGrant != nil && existingGrant.GetAllowance() != nil {
+	existingGrant, err := k.getGrant(ctx, granter, grantee)
+	if err != nil && !strings.Contains(err.Error(), "fee-grant not found") {
```

Comment:
Checking against error contents is fragile. Can we instead use `Is` or `As`?

Reply:
thanks for pointing out, I used this to make sure the tests pass, forgot to change after that.